app/src/main/java/com/mobile/tandil/javabaseproject/activity/MainActivity.java

```diff
@@ -13,13 +13,15 @@
 public class MainActivity extends AppCompatActivity {
 
     private ParkingContract.Presenter presenter;
+    private ActivityMainBinding binding;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_main);
+        binding = ActivityMainBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
 
-        presenter = new ParkingPresenter(new ParkingModel(), new ParkingView(this));
+        presenter = new ParkingPresenter(new ParkingModel(), new ParkingView(this, this.binding));
+        binding.buttonMainActivity.setOnClickListener(view -> presenter.onShowButtonPressed());
```

Comment:
you should put this line, and all the listeners, in a separated method called setListeners or something like that

Reply:
done